src/_01_IntroToArrayLists/_06_IPodShuffle.java

```diff
@@ -1,27 +1,68 @@
 package _01_IntroToArrayLists;
 
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
 import java.util.ArrayList;
 import java.util.Random;
 
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
 
 //Copyright The League of Amazing Programmers, 2015
 
-public class _06_IPodShuffle{
+public class _06_IPodShuffle implements ActionListener {
+	JFrame frame = new JFrame();
+	JPanel panel = new JPanel();
+	JButton playButton = new JButton();
+
+	ArrayList<Song> songsList = new ArrayList<>();
+
+	Random rand = new Random();
+	int randomNum;
+	int currentSong;
+
 	public _06_IPodShuffle() {
-		// 1. Use the Song class the play the demo.mp3 file.
-				
-				
-		/**
-		 * 2. Congratulations on completing the sound check! * Now we want to make an
-		 * iPod Shuffle that plays random music. * Create an ArrayList of Songs and a
-		 * "Surprise Me!" button that will play a random song when it is clicked. * If
-		 * you're really cool, you can stop all the songs, before playing a new one on
-		 * subsequent button clicks.
-		 */
-		
+		songsList.add(new Song("ipod-dramatic.mp3"));
```

Comment:
We haven't learned why yet, but it's best practice to limit what we do in the constructor.  Try to just use the constructor to initialize instance variables from the parameters that were passed in.  You can put initialization code, like this, in a method.